src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -35,6 +36,17 @@ namespace NuGet.PackageManagement.VisualStudio
     public abstract class PackageReferenceProject<T, U> : BuildIntegratedNuGetProject, IPackageReferenceProject where T : ICollection<U>, new()
     {
         internal static readonly Comparer<PackageReference> PackageReferenceMergeComparer = Comparer<PackageReference>.Create((a, b) => a?.PackageIdentity?.CompareTo(b.PackageIdentity) ?? 1);
+        private static readonly Lazy<bool> IsCounterFactualTriggered = new(() =>
```

Comment:
Whenever adding a lazy in the VS world, it's worth considering whether the Lazy function may require the UI thread. 

If it does, and the UI thread is blocked on a thread calling this lazy, it could deadlock.

Reply:
This is _not_ supposed to run in UI thread. But, sometimes, this runs on UI thread :D

Emitting telemetry does not require UI thread.

I don't need the lazy value. I just want to run this once, just before getting experiment flag. Is it worth changing to  https://docs.microsoft.com/en-us/dotnet/api/system.threading.lazyinitializer.ensureinitialized?view=netframework-4.7.2 ?

Reply:
Yes, lazy initializer would have the same result. 
The challenge is that we don't know own the telemetry infrastructure, so we don't really know whether it'll require the UI thread.

A safer approach might be to have 1 thread check a value rather than emitting the telemetry in the lazy.

Reply:
Update: we just chatted with experiments team and we can emit as many counterfactuals as we need.